submissions/the-darkesttt/frogger-task/app.js

```diff
@@ -0,0 +1,323 @@
+        this.bg.classList.add('bg');
+        this.menu = document.createElement('div');
+        this.menu.classList.add('menu');
+        this.menuTitle = document.createElement('h1');
+        this.menuTitle.innerText = 'Game Menu';
+        this.menuDescriprion = document.createElement('p');
+    }
+
+    start() {
+
+        this.menuDescriprion.innerText = 'Choose your character!';
+        this.menuList = document.createElement('ul');
+
+        playerSkins.forEach((item) => {
+            this.menuItem = document.createElement('li');
+    
+            const itemImg = document.createElement('img');
+            const imgSrc = item.src;
+            const imgId = item.id;
+            itemImg.setAttribute('src', imgSrc);
+            itemImg.setAttribute('id', imgId);
+            const imgDesc = document.createElement('p');
+            imgDesc.innerText = imgId;
+    
+            this.menuList.appendChild(this.menuItem);
+            this.menuItem.appendChild(itemImg);
+            this.menuItem.appendChild(imgDesc);
+            
+        });
+
+        document.body.appendChild(this.bg);
+        document.body.appendChild(this.menu);
+
+        this.menu.appendChild(this.menuTitle);
+        this.menu.appendChild(this.menuDescriprion);
+        this.menu.appendChild(this.menuList);
+
+        this.menuList.addEventListener('click', (event) => {
+            const listTarget = event.target;
+            if (listTarget.tagName == "UL") return;
+        
+            player.sprite = listTarget.getAttribute("src");
+    
+            this.bg.classList.add('menu-animation');
+            this.menu.classList.add('menu-animation');
+
+        });
+    }
+
+    death(points) {
+        if (this.menuList.parentElement != null) {
+            this.menuList.parentElement.removeChild(this.menuList);
+        }
+
+        this.bg.classList.remove('menu-animation');
+        this.menu.classList.remove('menu-animation');
+
+        this.menuDescriprion.innerText = 'You died! Your score:';
+
+        this.menuScore = document.createElement('span');
+        this.menuScore.innerText = points;
+        this.menuScore.classList.add('menu-score');
+
+        this.menu.appendChild(this.menuScore);
+
+        this.menuClose = document.createElement('button');
+        this.menuClose.innerText = 'Close menu and continue';
+        this.menuClose.classList.add('menu-close');
+        this.menu.appendChild(this.menuClose);
+
+        this.menuClose.addEventListener('click', (event) => {
+            this.bg.classList.add('menu-animation');
+            this.menu.classList.add('menu-animation');
+
+            this.menuDescriprion.innerText = '';
+            this.menuScore.parentElement.removeChild(this.menuScore);
+            this.menuClose.parentElement.removeChild(this.menuClose);
+        });
+    }
+}
+
+
+const startMenu = new Menu();
+startMenu.start();
+
+let points = 0;
+const score = document.createElement('p');
```

Comment:
Better names: `score` kinda sounds like `points`, though it doesn't store value, it stores an element. Maybe `scoreElem` could be better.